plugins/content/createmenulink/createmenulink.php

```diff
@@ -0,0 +1,215 @@
+						);
+
+			if (!empty($menuItems))
+			{
+				$data->menuid    = $menuItems[0]->id;
+				$data->menutitle = $menuItems[0]->title;
+				$data->menualias = $menuItems[0]->alias;
+				$data->menutype  = $menuItems[0]->menutype;
+				$data->parent_id = $menuItems[0]->parent_id;
+			}
+
+			if (empty($menuItems))
+			{
+				JHtml::_('jquery.framework', false);
+				$document = JFactory::getDocument();
+				$document->addScriptDeclaration('
+				(function ($) {
+                    			$(document).ready(function() {
+                        			$("#jform_title").on("keyup", function() {
+                            				$("#jform_menutitle").val($(this).val());
+                        			});
+                        
+                        			$("#jform_name").on("keyup", function() {
+                            				$("#jform_menutitle").val($(this).val());
+                        			});
+
+                        			$("#jform_alias").on("keyup", function() {
+                            				$("#jform_menualias").val($(this).val());
+                        			});
+                    			});
+				}(jQuery));
+				');
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Adds additional fields to the editing form
+	 *
+	 * @param   JForm  $form  The form to be altered.
+	 * @param   mixed  $data  The associated data for the form.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   3.6
+	 */
+	public function onContentPrepareForm($form, $data)
+	{
+		if (!($form instanceof JForm))
+		{
+			$this->_subject->setError('JERROR_NOT_A_FORM');
+
+			return false;
+		}
+
+		// Check we are manipulating a valid form.
+		$name = $form->getName();
+
+		if (!in_array($name, array($this->params->get('componentview'))))
+		{
+			return true;
+		}
+
+		$app = JFactory::getApplication();
+
+		if ($app->isAdmin())
+		{
+			// Add the fields to the form.
+			JForm::addFormPath(__DIR__ . '/forms');
+			$form->loadFile('createmenulink', false);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Sets the fields value from edit form
+	 *
+	 * @param   JForm  $context  The form to be altered.
+	 * @param   mixed  $article  The associated data for the form.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   3.6
+	 */
+	public function onContentBeforeSave($context, $article)
+	{
+		$session = JFactory::getSession();
+		$session->set("formData", JFactory::getApplication()->input->post->get('jform', array(), 'array'));
+
+		return true;
+	}
+
+	/**
+	 * Adds additional fields to the editing form
+	 *
+	 * @param   JForm  $context  The form to be altered.
+	 * @param   mixed  $article  The associated data for the form.
+	 *
+	 * @return  boolean
+	 *
+	 * @since   3.6
+	 */
+
+	public function onContentAfterSave($context, $article)
+	{
+		if (in_array($context, array('com_menus.item')))
+		{
+			return true;
+		}
+
+		$session = JFactory::getSession();
+		$data    = $session->get("formData");
+		$session->clear("formData");
+
+		list($component, $view) = explode('.', $context);
```

Comment:
You can get the option and view directly from `JFactory::getApplicaton()->input`. Which you should prefer in this case (for security and there are good chances people don't set the context always right - you don't even check if it contains a dot or is set at all).